common/utils/deprovision_statefulset.yml

```diff
@@ -8,87 +8,157 @@
+          replace:
+            path: "{{ app_deployer }}"
+            regexp: "rkey: rvalue"
+            replace: "{{ app_rkey }}: {{ app_rvalue }}"
+
+      when: app_replica is defined
+
+    - block:
+
+        - name: Get the application label values from env
+          set_fact:
+            app_lkey: "{{ app_label.split('=')[0] }}"
+            app_lvalue: "{{ app_label.split('=')[1] }}"
+
+        - name: Replace the application label placeholder
+          replace:
+            path: "{{ app_deployer }}"
+            regexp: "lkey: lvalue"
+            replace: "{{ app_lkey }}: {{ app_lvalue }}"
+
+      when: app_label is defined
+
+    - name: Delete the application and its related service.
+      shell: kubectl delete -f {{ app_deployer }} -n {{ app_ns }}
+      register: app_status
+      until: 'app_status.rc == 0'
+      delay: 30
+      retries: 10
+
+    - name: Deleting the PVC
+      shell: kubectl delete pvc {{ item }} -n {{ app_ns }}
+      args:
+        executable: /bin/bash
+      with_items:
+        - "{{ pvc_list.stdout_lines }}"
+
+    - name: Check if the PVCs are deleted
+      shell: kubectl get pvc -n {{ app_ns }}
+      register: list_pvc
+      until: "'No resources found' in list_pvc.stderr"
+      delay: 30
+      retries: 15
+
+    - block:
+
+        - name: Check if the ctrl-pod is deleted.
+          shell: >
+             kubectl get pods -n {{ app_ns }}
+             -l openebs.io/controller=jiva-controller --no-headers 
+          args:
+            executable: /bin/bash
+          register: ctrl_status     
+          until: "'No resources found' in ctrl_status.stderr"
+          delay: 5
+          retries: 15
+            
+        - name: Check if the  replica-pod is deleted.
+          shell: kubectl get pods -n {{ app_ns }} -l openebs.io/replica=jiva-replica
+          args:
+            executable: /bin/bash
+          register: rep_status
+          until: "'No resources found' in rep_status.stderr"
+          delay: 5
+          retries: 15    
+
+      when: stg_engine == 'jiva'
+
+    - block:
+
+        - name: Check if the target pod is deleted.
+          shell: >
+              kubectl get pods -l openebs.io/target=cstor-target  -n {{ operator_ns }} --no-headers
+          args:
+            executable: /bin/bash     
+          register: target_status
+          until: " item.stdout  not  in target_status.stdout"
+          delay: 5
+          retries: 10
+          with_items:
+            - "{{ pv_list.results }}"      
+
+        - name: Check status of the pool deployments from cvr
+          shell: >
+              kubectl get cvr -n {{ operator_ns }} --no-headers
+          args:
+            executable:  /bin/bash
+          register: cvr_status
+          until: " item.stdout not  in cvr_status.stdout"
+          delay: 5
+          retries: 10
+          with_items: 
+            - "{{ pv_list.results }}"    
+
+      when: stg_engine == 'cstor'
```

Comment:
Can we check if all the application deployment litmus jobs has this 'stg_engine' defined?